src/chainstate/stacks/index/profile.rs

```diff
@@ -0,0 +1,409 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2022 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::time::SystemTime;
+
+/// Fine-grained profiling data for Trie storage ops
+#[derive(Debug, Clone)]
+pub struct TrieBenchmark {
```

Comment:
Is there a no-op version of this that is used in production, or will all this benchmarking code be invoked in all release builds?

Reply:
I can add a no-op version for production.  It's currently only used for testing.